fedbiomed/researcher/experiment.py

```diff
@@ -1574,12 +1574,18 @@ def run_once(self, increase: bool = False, test_after: bool = False) -> int:
         aggr_args_thr_msg, aggr_args_thr_file = self._aggregator.create_aggregator_args(self._global_model,
                                                                                         self._job.nodes)
 
+        # Collect auxiliary variables from the aggregates optimizer, if any.
+        optim_aux_var = self._collect_optim_aux_var()
+
         # Trigger training round on sampled nodes
-        _ = self._job.start_nodes_training_round(round_=self._round_current,
-                                                 aggregator_args_thr_msg=aggr_args_thr_msg,
-                                                 aggregator_args_thr_files=aggr_args_thr_file,
-                                                 do_training=True,
-                                                 secagg_arguments=secagg_arguments)
+        self._job.start_nodes_training_round(
+            round_=self._round_current,
+            aggregator_args_thr_msg=aggr_args_thr_msg,
+            aggregator_args_thr_files=aggr_args_thr_file,
+            do_training=True,
+            secagg_arguments=secagg_arguments,
+            optim_aux_var=optim_aux_var,
+        )
 
         # refining/normalizing model weights received from nodes
         model_params, weights, total_sample_size, encryption_factors = self._node_selection_strategy.refine(
```

Comment:
I know this is not your doing but for the line `aggregated_params: Dict[str, Union['torch.tensor', 'nd.ndarray']] = \`
should it be `np.ndarray` instead?

Reply:
It should really be `Dict[str, Union[torch.Tensor, np.ndarray]]`. There is absolutely no value in not importing numpy and torch here, since they are automatically imported by `fedbiomed.common.training_plans`. Indeed, this is pre-existing, but I am all in favor of Uncle Bob's Boy Scout Rule, so I will happily change that.